packages/rocketchat-livechat/imports/server/rest/visitors.js

```diff
@@ -30,3 +30,31 @@ RocketChat.API.v1.addRoute('livechat/visitor/:visitorToken/room', { authRequired
 		return RocketChat.API.v1.success({ rooms });
 	}
 });
+
+RocketChat.API.v1.addRoute('livechat/visitor/:visitorToken/room/transfer', { authRequired: true }, {
+	post() {
+		if (!RocketChat.authz.hasPermission(this.userId, 'view-livechat-manager')) {
+			return RocketChat.API.v1.unauthorized();
+		}
+
+		const rooms = RocketChat.models.Rooms.findOpenByVisitorToken(this.urlParams.visitorToken).fetch();
+		if (!rooms || rooms.length === 0) {
+			return RocketChat.API.v1.failure('Visitor dont have any opened room');
+		}
+		const room = rooms[0];
+
+		const visitor = LivechatVisitors.getVisitorByToken(this.urlParams.visitorToken);
+		if (!visitor) {
+			return RocketChat.API.v1.failure('Visitor not found');
+		}
+
+		return RocketChat.API.v1.success({
+			transfered: RocketChat.Livechat.transfer(room, visitor, {
+				userId: this.bodyParams.userId,
```

Comment:
I have some concerns about passing `userId` by parameter and not using the logged-in user (` this.userId`).